plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -948,6 +937,8 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         }
         RewriteFiles rewriteFiles = transaction.newRewrite();
         rewriteFiles.rewriteFiles(scannedFiles, newFiles);
+        optimizeHandle.getSnapshotId().map(icebergTable::snapshot)
```

Comment:
`.map` will silently mask the fact that `table.snapshot(long id)` may return null for an invalid snapshot

> Get the snapshot of this table with the given id, or null if there is no matching snapshot.


**at least**, we should use `.flatMap` + `Optional.of` + comment